service/s3/internal/awsrestxml/marshal.go

```diff
@@ -60,35 +64,33 @@ func MarshalPutObjectInputShapeAWSREST(input *types.PutObjectInput, encoder *v2.
 	encoder.AddHeader("x-amz-storage-class").String(string(input.StorageClass))
 	encoder.AddHeader("x-amz-server-side-encryption").String(string(input.ServerSideEncryption))
 	if err := encoder.AddHeader("Expires").Time(*input.Expires, protocol.RFC822TimeFormatName); err != nil {
```

Comment:
Need to make sure to check for nil before setting any pointer based value.

For enums need to see if the enum value is not empty string.